libraries/AP_Proximity/AP_Proximity_AirSimSITL.cpp

```diff
@@ -55,29 +57,23 @@ void AP_Proximity_AirSimSITL::update(void)
                 }
             } else {
                 // new sector started, previous one can be updated
-                _distance_valid[_last_sector] = true;
-                _angle[_last_sector] = _angle_deg_last;
-                _distance[_last_sector] = _distance_m_last;
-
+                boundary.mark_distance_valid(true, _last_sector);
+                boundary.set_angle(_angle_deg_last, _last_sector);
```

Comment:
we should probably replace, "set_angle" with "set_closest_point_in_sector".  the arguments would then be sector, yaw and pitch.

Reply:
we could probably remove the mark_distance_valid and instead have the bool set to true when any caller calls "set_distance".